example_scripts/spectrum_viewer.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+
+from __future__ import print_function
+
+import sys
+import os
+
+import pymzml
+
+import plotly.graph_objs as go
+
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+
+
+print('loading run...')
+run = pymzml.run.Reader(sys.argv[1])
+
+p = pymzml.plot.Factory()
+p.new_plot()
+
+tic_x=[]
+tic_y=[]
+for x,y in run['TIC'].peaks():
+    tic_x.append(x)
+    tic_y.append(y)
+max_tic = max(tic_y)
+
+
+app = dash.Dash(__name__)
+
+
+app.layout = html.Div([
+    dcc.Graph(
+        id='spectrum-plot',
+        style={
+            'fontFamily': 'Helvetica',
+        }
+    ),
+    html.H3(
+        children='Please specify spectrum ID',
+        style={
+            'fontFamily': 'Helvetica',
+        }
+    ),
+    dcc.Input(
+        id='spectrum-input-field',
+        value=1,
+        type='text',
+        style={
+            'width': '20%',
+            'height': '30px',
+            'lineHeight': '30px',
+            'borderWidth': '1px',
+            'borderStyle': 'dashed',
+            'borderRadius': '5px',
+            'fontFamily': 'Helvetica',
+            'fontSize': '130%'
+        }
+    ),
+    dcc.Graph(
+        id='tic-plot',
+    ),
+])
+
+spectra_buffer = {}
+
+def get_spectrum(spectrum_id):
+    if spectrum_id not in spectra_buffer.keys():
+        spectrum  = run[spectrum_id]
+        spectra_buffer[spectrum_id] = spectrum
+    else:
+        spectrum = spectra_buffer[spectrum_id]
+    return spectrum
+
+
+@app.callback(
+    dash.dependencies.Output('tic-plot', 'figure'),
+    [
+        dash.dependencies.Input('spectrum-input-field', 'value'),
+    ]
+)
+def update_TIC(spectrum_id_from_input=None):
+    if spectrum_id_from_input not in [None, '']:
+        spectrum_id  = int(spectrum_id_from_input)
+    else:
+        spectrum_id = 1
+    if spectrum_id > run.info['spectrum_count']:
```

Comment:
This is only able if the file is not truncated, maybe add a comment if someone runs into problems?

Reply:
I agree